Core/include/Acts/EventData/detail/DifferenceCalculator.hpp

```diff
@@ -0,0 +1,60 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2016-2020 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/EventData/detail/ParameterTraits.hpp"
+
+#include <utility>
+
+#include <Eigen/Core>
+
+namespace Acts {
+namespace detail {
+
+/// Compute the minimal difference between parameters within the nominal range.
+///
+/// @tparam indices_t Parameter indices enum type
+/// @tparam kParameters Parameter indices pack that defines the used parameter
+template <typename indices_t, indices_t... kParameters>
+struct DifferenceCalculator {
```

Comment:
Why not `ResidualCalculator`?

Reply:
Nothing in the implementation requires it to be a residual. Depending on who you ask, the residual between `a` and `b` might be either `a - b` or `b - a`. To avoid this confusion as well, this just acts as a binary operator `diff(a, b) = a - b` and which input is which is up to the caller.